llvm/lib/Target/Xtensa/XtensaISelLowering.cpp

```diff
@@ -594,6 +594,26 @@ SDValue XtensaTargetLowering::LowerSELECT_CC(SDValue Op,
                      FalseValue, TargetCC);
 }
 
+SDValue XtensaTargetLowering::LowerRETURNADDR(SDValue Op,
+                                              SelectionDAG &DAG) const {
+  // check the depth
+  // TODO: xtensa-gcc can handle this, by navigating through the stack, we
+  // should be able to do this too
+  assert((cast<ConstantSDNode>(Op.getOperand(0))->getZExtValue() == 0) &&
```

Comment:
Shouldn't assert on this, it's not invalid IR. Langref states it returns "zero if it cannot be identified"

Reply:
Fixed. I hope  that I understand correctly your comment,